ci: add msrv check job

king-11 opened the pull request:
related to #1520

king-11 committed:
ci: add msrv check job

Signed-off-by: Lakshya Singh <lakshay.singh1108@gmail.com>

niklasad1 commented on .github/workflows/ci.yml:
```diff
@@ -100,6 +100,29 @@ jobs:
         run: cargo machete
 
+  check-msrv:
+    name: Check
```
```suggestion
    name: Check MSRV
```

niklasad1 committed:
Update .github/workflows/ci.yml

jsdw commented on .github/workflows/ci.yml:
```diff
@@ -100,6 +100,29 @@ jobs:
       - name: Check unused dependencies
         run: cargo machete
 
+  check-msrv:
```
Am I missing how this is checking MSRV? It looks like all it does is run `carrgo test`, same as the below workflow.

jsdw commented on .github/workflows/ci.yml:
```diff
@@ -100,6 +100,29 @@ jobs:
+        uses: actions-rs/toolchain@v1.0.7
+        with:
+          profile: minimal
+          toolchain: 1.74.0
```
I missed this at first so maybe just to make it obvious:

```suggestion
          # Check that tests pass using our MSRV:
          toolchain: 1.74.0
```

king-11 replied:
updated

niklasad1 reviewed (commented):
Looks overall good but the UI tests won't pass for all rustc compilers down to 1.74.0

Either just type-check the code with 1.74.0 or disable the tests in the `proc-macro-crate`.

king-11 commented:
@niklasad1 what do you mean by type check?

niklasad1 commented:
> @niklasad1 what do you mean by type check? As in build?

`cargo check --all-targets` but cargo build would work as well doesn't matter to me

king-11 committed:
ci: replace test with check all targets

UI tests won't pass for all rustc compilers down to 1.74.0 so just type
checking code for the particular toolchain

Signed-off-by: Lakshya Singh <lakshay.singh1108@gmail.com>

king-11 commented:
should be good now sorry for delay was in phase of changing cities
